test/jit/test_list_dict.py

```diff
@@ -1288,6 +1288,14 @@ def update(a, b):
         self.checkScript(update, (self.dict(), self.dict()))
         self.checkScript(update, (self.dict(), self.dict2()))
 
+        def upd() -> Dict[str, int]:
+            a: Dict[str, int] = {}
+            for i in range(3):
+                a.update({'a': i})
```

Comment:
Do we need several iterations? Would something like this suffice:
```suggestion
            a: Dict[str, int] = {'a': 0}
            a.update({'a': 1})
```